### DISCUSSION TITLE
How to determine the OS type at compile time

### DISCUSSION THREAD

**Discussion by tsctsc6** [AvaloniaUI/Avalonia#16399] [Q&A] [answered]
Let's say I have a xplat project.
As we know, the project has a core library that is called by other startup projects.
How to determine the OS type at compile time in the core library?
The reason is that sometimes I want to call different OS interfaces under different OS.

**Comment by maxkatz6** [comment, marked as answer]
You can use standard OperatingSystem APIs. https://learn.microsoft.com/en-us/dotnet/api/system.operatingsystem?view=net-8.0

Ideally, use `OperatingSystem.IsWindows()` or `OperatingSystem.IsLinux()` methods, as these help with trimming code from if you build your code for a specific runtime identifier.

**Reply by tsctsc6** [reply]
Thanks.
